src/py/flwr/server/strategy/dp_adaptive_clip_strategy.py

```diff
@@ -0,0 +1,80 @@
+"""DP-FedAvg [McMahan et al., 2018] strategy.
+
+Paper: https://arxiv.org/pdf/1710.06963.pdf
+"""
+
+
+import math
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+
+from flwr.common import FitRes, Parameters, Scalar
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy.dp_fixed_clip_strategy import DPFixedClipStrategy
+from flwr.server.strategy.strategy import Strategy
+
+
+class DPAdaptiveClipStrategy(DPFixedClipStrategy):
+    """Wrapper for configuring a Strategy for DP with Adaptive Clipping."""
+
+    # pylint: disable=too-many-arguments,too-many-instance-attributes
+    def __init__(
+        self,
+        strategy: Strategy,
+        num_sampled_clients: int,
+        noise_multiplier: float = 1,
+        init_clip_norm: float = 0.1,
+        clip_norm_lr: float = 0.2,
+        clip_norm_target_quantile: float = 0.5,
+        clip_count_stddev: Optional[float] = None,
+    ) -> None:
+
+        super().__init__(
+            strategy, num_sampled_clients, noise_multiplier, init_clip_norm
+        )
+        self.clip_norm_lr = clip_norm_lr
+        self.clip_norm_target_quantile = clip_norm_target_quantile
+        if not clip_count_stddev:
+            self.clip_count_stddev = (
+                self.num_sampled_clients / 20.0 if noise_multiplier else 0
```

Comment:
Where does the `20.0` come from?

Reply:
It's a recommendation from the [paper](https://arxiv.org/pdf/1905.03871.pdf) that the algorithm is based on.

Reply:
Would it make sense to expose this as a configurable parameter?

Reply:
`clip_count_stddev` is already exposed as a configurable parameter that can be passed in to the constructor, am i misunderstanding something?

Reply:
Ah, sorry, you're obviously right, I was just looking at the `20.0` and I totally ignored the larger context.